apps/asap-server/test/controllers/research-outputs.test.ts

```diff
@@ -609,6 +617,61 @@ describe('ResearchOutputs controller', () => {
       });
     });
   });
+
+  describe('create', () => {
+    afterEach(() => {
+      expect(nock.isDone()).toBe(true);
+    });
+
+    afterEach(() => {
+      nock.cleanAll();
+    });
+    test('creating the research output should return the id from squidex rest', async () => {
+      const researchOutput = { type: 'test' } as unknown as ResearchOutput;
+      const teamId = 'team-id-1';
+      const researchOutputId = 'created-output-id';
+      nock(config.baseUrl)
+        .post(`/api/content/${config.appName}/research-outputs?publish=false`, {
+          type: { iv: 'test' },
+          asapFunded: { iv: 'Not Sure' },
+          usedInAPublication: { iv: 'Not Sure' },
+        })
+        .reply(201, { id: researchOutputId })
+        .get(`/api/content/${config.appName}/teams/${teamId}`)
+        .matchHeader('X-Unpublished', `true`)
+        .reply(200, { data: { id: teamId, outputs: { iv: ['output-1'] } } })
+        .patch(`/api/content/${config.appName}/teams/${teamId}`, {
+          outputs: { iv: ['output-1', researchOutputId] },
+        })
+        .reply(200, getUpdateTeamResponse()); // response is not used
+
+      const id = await researchOutputs.create(researchOutput, teamId);
+      expect(id).toEqual({ id: researchOutputId });
+    });
+    test('should throw when fails to create the research output - 400', async () => {
+      const teamId = 'team-id-1';
+      const researchOutput = { type: 1 } as unknown as ResearchOutput;
+      nock(config.baseUrl)
+        .post(`/api/content/${config.appName}/research-outputs?publish=false`)
+        .reply(400);
+
+      await expect(
+        researchOutputs.create(researchOutput, teamId),
+      ).rejects.toThrow('Bad Request');
+    });
+
+    test('should throw when fails to create the research output - 500', async () => {
```

Comment:
add a test to show what happens when team association fails maybe?

Reply:
we are going to need it later anyway when we add some sort of level of atomicity

Reply:
done